### PR TITLE
Internal conventions implementations expose their public type

### PR DESCRIPTION
**Pull request by eskatos**
For the Kotlin DSL provider to generate conventions accessors that use non-internal types.
See https://github.com/gradle/kotlin-dsl/issues/343

### CODE REVIEW

**Comment by eskatos** [comment]
Feedback build https://builds.gradle.org/viewQueued.html?itemId=12675141

File: subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/internal/AntlrSourceVirtualDirectoryImpl.java [hunk 1]
```diff
@@ -51,4 +53,9 @@ public AntlrSourceVirtualDirectory antlr(Action<? super SourceDirectorySet> conf
         configureAction.execute(getAntlr());
         return this;
     }
+
+    @Override
+    public TypeOf<?> getPublicType() {
+        return TypeOf.typeOf(AntlrSourceVirtualDirectory.class);
```
Comment (bamboo) [inline subprojects/antlr/src/main/java/org/gradle/api/plugins/antlr/internal/AntlrSourceVirtualDirectoryImpl.java]: :o: Statically import `TypeOf.typeOf` here and elsewhere to improve prose?

**Comment by eskatos** [comment]
Let's try again https://builds.gradle.org/viewQueued.html?itemId=12701130

**Comment by bamboo** [comment]
And again: https://builds.gradle.org/viewQueued.html?itemId=12706036&tab=queuedBuildOverviewTab

**Comment by bamboo** [comment]
Master Accept (Trigger): https://builds.gradle.org/viewQueued.html?itemId=12710199&tab=queuedBuildOverviewTab

**Pull request merged**